backend/src/main/kotlin/com/ci5644/trade/services/user/UserService.kt

```diff
@@ -1,7 +1,53 @@
 package com.ci5644.trade.services.user
 
-import com.ci5644.trade.dto.UserDto
+import com.ci5644.trade.repositories.UserRepository
+import com.ci5644.trade.models.user.UserEntity
+import com.ci5644.trade.dto.UserDetailsDto
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import org.springframework.security.core.userdetails.User
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.security.core.userdetails.UserDetailsService
+import org.springframework.security.core.userdetails.UsernameNotFoundException
 
-interface UserService {
-    fun createUser(userDto: UserDto): UserDto
+/**
+ * Service class for managing user-related operations.
+ */
+@Service
+class UserService : UserDetailsService {
+
+    @Autowired
+    lateinit var userRepository: UserRepository
+
+    /**
+     * Loads user details by username.
+     * @param username String - The username of the user.
+     * @return UserDetails - User details.
+     * @throws UsernameNotFoundException if the user with the specified username is not found.
+     */
+    @Throws(UsernameNotFoundException::class)
+    override fun loadUserByUsername(username: String): UserDetails {
+        if (!userRepository.existsByUsername(username))
```

Comment:
you are making two calls to the db. findbyusername should return optional and you can do ?: in the end and throw the exception